### ISSUE TITLE
graph not working on windows nanoserver

### ISSUE THREAD

**Issue by kobellem** [docker/docker#26604]
**Description**
Running containers on azure windows nanoserver.

C drive ran out of space, so I used the graph config option to store the images on the D drive by adding "graph": "D:\Programfiles\docker" to daemon.json.

**Steps to reproduce the issue:**
1. Install containers on windows nanoserver.
2. Change the drive to store the images by use of the graph config option.
3. pull microsoft/nanoserver image.

**Describe the results you received:**
failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer D:\programdata\docker\windowsfilter\9a5896fda9f59ec25cdda8ef9d40969a6215bf1072965cf2290a2831cf347384: The request is not supported.

**Describe the results you expected:**
Image stored in D:\Programdata\docker

**Additional information you deem important (e.g. issue happens only occasionally):**
Connected with OSX client with tls verification.

**Output of `docker version`:**

```
$docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      windows/amd64
```

**Output of `docker info`:**

```
$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.0
Storage Driver: windowsfilter
 Windows: 
Logging Driver: json-file
Plugins:
 Volume: local
 Network: nat null overlay
Swarm: inactive
Security Options:
Kernel Version: 10.0 14300 (14300.1045.amd64fre.rs1_release_svc.160705-1059)
Operating System: Windows Server 2016 Datacenter Technical Preview 5
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 7 GiB
Name: ***
ID: RCCE:KXUE:75WR:3PNN:HZBW:2NVY:VWXZ:NPPC:M5FA:HMPS:RLEB:5AZ6
Docker Root Dir: D:\programdata\docker
Debug Mode (client): false
Debug Mode (server): false
Username: ***
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Comment by mlaventure** [comment]
ping @jhowardmsft

**Comment by jhowardmsft** [comment]
Is `D:` ReFS or NTFS?

**Comment by jhowardmsft** [comment]
https://github.com/docker/docker/pull/25509

**Comment by kobellem** [comment]
It's NTFS
> Get-Volume

DriveLetter FileSystemLabel   FileSystem DriveType HealthStatus OperationalStatus SizeRemaining  Size
----------- ---------------   ---------- --------- ------------ ----------------- -------------  ----
D           Temporary Storage NTFS       Fixed     Healthy      OK                     12.15 GB 14 GB
C                                           NTFS       Fixed     Healthy      OK                      1.87 GB  8 GB

**Comment by jhowardmsft** [comment]
Curious. The Windows to Windows CI servers in Azure have run with a second drive (SSD) since pre-TP4 with no issues at all, so this sounds environmental. Although they don't run as a service. Can you try using docker master (there are many fixes since 1.12), and attempt running directly from the command line such as `dockerd -g d:\daemon` to see if it is specific to running as a service.

**Comment by jhowardmsft** [comment]
Closing as no answer to above question for almost 6 months.

**Comment by dermeister0** [comment]
The graph option does not work for me on Windows 10.

I totally uninstalled Docker and removed C:\ProgramData\Docker directory. Then I installed the Docker Edge CE.

```
> docker --version
Docker version 17.04.0-ce, build 4845c56
```

I opened Settings->Daemon window and edited configuration:

```
{
  "registry-mirrors": [],
  "insecure-registries": [],
  "debug": true,
  "experimental": true,
  "graph": "D:\\Docker2"
}
```

After Docker restart, I pulled an image.

```
docker image pull microsoft/nanoserver
```

Expected result:
D:\Docker2 directory is created. Image is saved there.

Actual result:
Image is saved in C:\ProgramData\Docker.
